app.R

```diff
@@ -1830,6 +1830,19 @@ select count(nct_id) as number_sites, nct_id from trial_sites where org_status =
       df_crit$biomarker_api_inc_matches <- NA 
     }
     #browser()
+
+    # Load prior therapy previously saved by ETL job.
+    trials_with_prior_therapy_inc_df <- safe_query(dbGetQuery,
+      "select nct_id, nci_thesaurus_concept_id from trial_prior_therapies where eligibility_criterion='inclusion'")
+    trials_with_prior_therapy_exc_df <- safe_query(dbGetQuery,
+      "select nct_id, nci_thesaurus_concept_id from trial_prior_therapies where eligibility_criterion='exclusion'")
+
+    # TODO (jcallaway): save nci_thesaurus_concept_id here so we can use it later to match
+    # against the patient's prior therapies, insterad of just tagging trials if they have
+    # any inclusion/exclusion as I am doing here.
+    df_crit <- transform(df_crit, pt_inc_matches = ifelse(clean_nct_id %in% trials_with_prior_therapy_inc_df$nct_id, TRUE, FALSE))
+    df_crit <- transform(df_crit, pt_exc_matches = ifelse(clean_nct_id %in% trials_with_prior_therapy_exc_df$nct_id, TRUE, FALSE))
+
```

Comment:
These need to be functional applys -- and perhaps with a function that returns three values  0 for false, 1 for true, and .5 for the maybes in the inclusion sense.  

I..e check_if_any(first TRIAL code in prior therapy inclusion) || check if any (second TRIAL code in pt inclusion || ... || ) -- if that is TRUE, then it would return TRUE.  
ELSE (i.e. none of those return TRUE) 
     -- Turn the sense of the checking around....  Have an environment of the criteria codes, and call the check_if_anyon the stuff in patient data environment NOTE -- can skip this for the moment to get the simple case working.
ELSE FALSE